operator/src/main/java/org/keycloak/operator/controllers/KeycloakUpdateJobDependentResource.java

```diff
@@ -0,0 +1,190 @@
+        super(Job.class);
+        this.configureWith(new KubernetesDependentResourceConfigBuilder<Job>()
+                .withLabelSelector(Constants.DEFAULT_LABELS_AS_STRING)
+                .build());
+    }
+
+    @Override
+    protected Job desired(Keycloak primary, Context<Keycloak> context) {
+        var builder = new JobBuilder();
+        builder.withMetadata(createMetadata(jobName(primary), primary));
+        var specBuilder = builder.withNewSpec();
+        addPodSpecTemplate(specBuilder, primary, context);
+        // we don't need retries; we use exit code != 1 to signal the upgrade decision.
+        specBuilder.withBackoffLimit(JOB_RETRIES);
+        // Remove the job after 30 minutes.
+        // TODO make it configurable!?
+        specBuilder.withTtlSecondsAfterFinished(JOB_TIME_TO_LIVE_SECONDS);
+        specBuilder.endSpec();
+        return builder.build();
+    }
+
+    public static boolean isJobFromCurrentKeycloakCr(Job job, Keycloak keycloak) {
+        var annotations = job.getMetadata().getAnnotations();
+        var hash = annotations.get(KEYCLOAK_CR_HASH_ANNOTATION);
+        return Objects.equals(hash, keycloakHash(keycloak));
+    }
+
+    public static String jobName(Keycloak keycloak) {
+        return keycloak.getMetadata().getName() + "-update-job";
+    }
+
+    private static String podName(Keycloak keycloak) {
+        return keycloak.getMetadata().getName() + "-update-pod";
+    }
+
+    private static ObjectMeta createMetadata(String name, Keycloak keycloak) {
+        var builder = new ObjectMetaBuilder();
+        builder.withName(name)
+                .withNamespace(keycloak.getMetadata().getNamespace())
+                .withLabels(Utils.allInstanceLabels(keycloak))
+                .withAnnotations(Map.of(KEYCLOAK_CR_HASH_ANNOTATION, keycloakHash(keycloak)));
+        return builder.build();
+    }
+
+    private static void addPodSpecTemplate(JobSpecFluent<?> builder, Keycloak keycloak, Context<Keycloak> context) {
+        var podTemplate = builder.withNewTemplate();
+        podTemplate.withMetadata(createMetadata(podName(keycloak), keycloak));
+        podTemplate.withSpec(createPodSpec(context));
+        podTemplate.endTemplate();
+    }
+
+    private static PodSpec createPodSpec(Context<Keycloak> context) {
+        var builder = new PodSpecBuilder();
+        builder.withRestartPolicy("Never");
+        addInitContainer(builder, context);
+        addContainer(builder, context);
+        builder.addNewVolume()
+                .withName(WORK_DIR_VOLUME_NAME)
+                .withNewEmptyDir()
+                .endEmptyDir()
+                .endVolume();
+        // For test KeycloakDeploymentTest#testDeploymentDurability
+        // it uses a pause image, which never ends.
+        // After 60 seconds, the job is terminated allowing the test to complete.
+        // TODO should be configurable?
+        builder.withActiveDeadlineSeconds(POD_DEADLINE_SECONDS);
```

Comment:
To be on the safe side, I'd rather go for 5 minutes. After all, two containers might need downloading, a re-augmentation might take place, etc. 

I wouldn't expose that to the users, unless we add a `updateCheckJobTemplate` in the `unspported` section. 
With 5 minutes being quite long, for testing it might be nice to override this using an environment variable like we do for the default Keycloak image that we pass to the operator.

Happy to discuss this further. Any more elaborate items can be a future PR.